### ISSUE TITLE
autoupdate is awesome except it gobbles comments and reformats lists :/

### ISSUE THREAD

**Issue by asottile** [pre-commit/pre-commit#210]
This seems promising: https://pypi.python.org/pypi/ruamel.yaml/0.6

Here's their example:

```
code = ruamel.yaml.load(inp, ruamel.yaml.RoundTripLoader)
code['name']['given'] = 'Bob'
print(ruamel.yaml.dump(code, Dumper= ruamel.yaml.RoundTripDumper), end='')
```

**Comment by Lucas-C** [comment]
+1 I've been bitten by this too :)

**Comment by elaichi** [comment]
+1

**Comment by asottile** [comment]
I'll likely merge #583 and call this "done" for now -- it uses some guesswork to replace sha lines intelligently while maintaining original formatting.

A different configuration language that actually roundtrips is really necessary to solve this in all cases.

**Comment by asottile** [comment]
via #583 -- available in 0.16.3
